deps/ncrypto/ncrypto.cc

```diff
@@ -1971,9 +1999,14 @@ int EVPKeyPointer::id(const EVP_PKEY* key) {
 #if OPENSSL_WITH_PQC
   // https://github.com/openssl/openssl/issues/27738#issuecomment-3013215870
   if (type == -1) {
-    if (EVP_PKEY_is_a(key, "ML-DSA-44")) return EVP_PKEY_ML_DSA_44;
-    if (EVP_PKEY_is_a(key, "ML-DSA-65")) return EVP_PKEY_ML_DSA_65;
-    if (EVP_PKEY_is_a(key, "ML-DSA-87")) return EVP_PKEY_ML_DSA_87;
+    const char* type_name = EVP_PKEY_get0_type_name(key);
+    if (type_name == nullptr) return -1;
+
+    for (const auto& mapping : pqc_mappings) {
+      if (strcmp(type_name, mapping.name) == 0) {
+        return mapping.nid;
+      }
+    }
```

Comment:
This seems fine if it works, but I'd appreciate if you could update the comment at the beginning of the highlighted section to better reflect why this lookup is done this way.